File: myblogsfrontend/src/app/service/user.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';
import { User } from '../payload/User';
import { Observable } from 'rxjs';
import { EditProfile } from '../payload/EditProfile';

@Injectable({
  providedIn: 'root'
})
export class UserService {

  // 'http://localhost:8080'
  host = environment.apiUrl;

  constructor(
    private http: HttpClient
  ) { }

  // create new user(customer signs up account)
  public createUser(user: User): Observable<User> {

    return this.http.post<User>(`${this.host}/user-create`, user);

  } // end of createUser()

  // update user profile
  public updateProfile(editProfile: EditProfile): Observable<User> {

    return this.http.put<User>(`${this.host}/edit-profile`, editProfile);

  } // end of updateProfile()

  // find user by user id
  findById(id: number): Observable<User> {

    return this.http.get<User>(`${this.host}/user-list/${id}`);

  } // end of findById()

} // end of class UserService

File: myblogsbackend/src/main/java/com/ez/myblogsbackend/service/UserService.java
package com.ez.myblogsbackend.service;

import com.ez.myblogsbackend.entity.User;
import com.ez.myblogsbackend.exception.BadDataException;
import com.ez.myblogsbackend.payload.EditProfile;

import javax.persistence.EntityNotFoundException;

public interface UserService {

    // find user by user id
    public User findById(Long id) throws EntityNotFoundException;

    // find user by email
    public User findUserByEmail(String email);

    // check whether user is inactive or not?
    // return:
    //  - not null: inactive user
    //  - null: not inactive user
    public User isInactiveUser(String email) throws BadDataException;

    // create new user(user signs up account)
    public User createUser(User user) throws BadDataException;

    // update user profile
    public User updateProfile(EditProfile editProfile) throws EntityNotFoundException;

}
